src/AbstractSQLCompiler.ts

```diff
@@ -103,15 +103,18 @@ export type CoalesceNode = [
 	UnknownTypeNodes,
 	...UnknownTypeNodes[],
 ];
+export type ToJSONNode = ['ToJSON', UnknownTypeNodes];
```

Comment:
According to postgres docs at least any element is allowed: https://www.postgresql.org/docs/14/functions-json.html#FUNCTIONS-JSON-CREATION-TABLE `to_json ( anyelement ) → json`
```suggestion
export type ToJSONNode = ['ToJSON', AnyTypeNodes];
```
where `AnyTypeNodes` is basically `Exclude<AbstractSqlType, 'string'>`, ie
```ts
export type AnyTypeNodes =
	| NullNode
	| DateNode
	| BooleanTypeNodes
	| NumberTypeNodes
	| TextTypeNodes
	| UnknownTypeNodes
	| DurationNode
	| SelectQueryNode
	| InsertQueryNode
	| UpdateQueryNode
	| DeleteQueryNode
	| UpsertQueryNode
	| SelectNode
	| ValuesNode
	| InnerJoinNode
	| LeftJoinNode
	| RightJoinNode
	| FullJoinNode
	| CrossJoinNode
	| GroupByNode
	| HavingNode
	| UnknownNode;
export type AbstractSqlType =
	| string
	| AnyTypeNodes;
```
Although I think `string` is only in `AbstractSqlType` because we don't yet have full typing for all the nodes so it's a bit of a catch-all to allow
```ts
export interface AbstractSqlQuery extends Array<AbstractSqlType> {
	0: string;
}
```
to work as a placeholder until we have full typing

Reply:
Alright, I'm started to see a pattern. I was struggling with how far I could/should tie things to Postgres docs as this is meant to support MySQL/SQLite as well.